### PR TITLE
Changed SchemaTable to SchemaViewer, Seperated imports in README

### LINKED ISSUE #22: Make Joi support tree-shakable
Right now, even if you don't use Joi, you end up pulling it.

Let's let tree-shaking omit that.  That means a separately importable component for the Joi schema viewer, probably just converting to json-schema and wrapping the "main" viewer component.

### PR DESCRIPTION
**Pull request by arshadkazmi42**
Fixes #22

### CODE REVIEW

**Comment by helfi92** [comment]
@arshadkazmi42 I separated the imports so that they become:

```jsx
import JoiSchemaTable from 'react-schema-viewer/JoiSchemaTable';
import SchemaTable from 'react-schema-viewer/SchemaTable';
```

Let me know what you think :)

**Comment by arshadkazmi42** [comment]
@helfi92 Actually I was following your inputs from #21 for import statements.
Is there any major deference in code performance between these two types of imports?
As, I feel these imports are better option, if there is no difference between performance
```js
import {JoiSchemaTable, SchemaTable} from 'react-schema-viewer';
```
As this import keeps it in single line and in first sight anyone can understand its getting imported from same place.

@djmitche any thoughts on this?

**Comment by helfi92** [comment]
https://github.com/taskcluster/react-schema-viewer/pull/26/commits/aa9fb89fdc5e9dc57f240b778cd239781415ed81 adds a way to also import via

```jsx
import { JoiSchemaTable, SchemaTable } from 'react-schema-viewer';
```

What's neat is that if the client consumer has tree-shaking set up properly, then they can use

```jsx
import { JoiSchemaTable } from 'react-schema-viewer';
```

otherwise they can simply import the individual component

```jsx
import JoiSchemaTable from 'react-schema-viewer/JoiSchemaTable';
```

**Review by djmitche — CHANGES_REQUESTED** [review]
I see that Neutrino magically "knows" to make an entry for each of the components, allowing the flexible imports.  Pretty cool!

I would like to also rename the class to SchemaViewer.  The README here is inconsistent with the code in the names of both Schema{Viewer,Table} and JoiSchema{Viewer,Table}.  I suspect that we soon will not be using a `<table>` element, at which point the name won't make much sense.

It's important to do the rename now, as we are now suggesting that users import things by name -- meaning that if we change the name later, it will be another breaking change.

**Comment by helfi92** [comment]
> I suspect that we soon will not be using a `<table>` element, at which point the name won't make much sense.


If we want this library to have the ability to display a graph in addition to the table, we could simply do something like

```jsx
import { SchemaGraph } from 'react-schema-viewer';`

// or
import SchemaGraph from 'react-schema-viewer/SchemaGraph';
```

> It's important to do the rename now, as we are now suggesting that users import things by name -- meaning that if we change the name later, it will be another breaking change.

I'm not sure why we would change the name later. Could you elaborate, please?

**Comment by djmitche** [comment]
I'm not sure what a schema "graph" would look like?  But I expect that this will probably be done with `<div>` elements and some careful positioning.  At any rate, the current situation is pretty weird: the project is named react-schema-viewer, and half of the README refers to SchemaViewer, while the code and the other half of the README refer to SchemaTable.  We should at least rationalize that, and since the project name is probably the hardest to change, and since we won't necessarily be using a table, it seems like SchemaViewer is the better choice.

**Comment by djmitche** [comment]
Well, "half" is wrong -- in the current master, the README is 100% SchemaViewer.  In this PR, it uses SchemaTable in all but one place.  I suppose we could fix that one place and rename the project/package to react-schema-table?

File: README.md [hunk 1]
```diff
@@ -9,33 +9,43 @@
...
-| `schema`                | Object                     | ✓         | Schema object representation in one the supported schema types.                                   |
-| `type`                  | enumerated: 'json' &#124; 'joi' | -         | Object schema validation type. Default: 'json'.                                                   |
+| `schema`                | Object                     | ✓         | A JSON schema object.                                                                             |
 | `headerBackgroundColor` | string                     | -         | The header background color given that a schema title is provided. Default: 'rgb(245, 245, 245)'. |
-| `maxHeight`             | string                     | -         | Max height of the panel. Default: '100%'.                                                          |
+| `maxHeight`             | string                     | -         | Max height of the panel. Default: '100%'.                                                         |
+
+## JoiSchemaViewer
```
Comment (djmitche) [inline README.md]: ..but this should be JoiSchemaTable :)

**Review by djmitche — APPROVED** [review]
Per some irc discussion, "viewer" is a general category for a bunch of related components, which happen to be table-like in appearance right now (but we could add other components someday).

@arshadkazmi42, apologies for taking over this PR!  Sometimes software development involves a lot of discussion and convincing :)

**Comment by djmitche** [comment]
(note I'm no longer suggesting renaming the package)

**Comment by helfi92** [comment]
This will be a breaking change.

**Comment by arshadkazmi42** [comment]
@djmitche Its great, that we came to an agreement 👍 
Thank you for involving so much in my PR

**Pull request merged**